userspace/engine/ruleset.h

```diff
@@ -106,18 +108,18 @@ class falco_ruleset
 
 		// Maps from event tag to a list of filters. There can
 		// be multiple filters for a given event tag.
-		std::vector<std::list<filter_wrapper *> *> m_filter_by_event_tag;
+		std::vector<std::unique_ptr<filter_wrapper_list>> m_filter_by_event_tag;
 
 	};
 
-	std::vector<ruleset_filters *> m_rulesets;
+	std::vector<std::unique_ptr<ruleset_filters>> m_rulesets;
 
 	// Maps from tag to list of filters having that tag.
-	std::map<std::string, std::list<filter_wrapper *>> m_filter_by_event_tag;
```

Comment:
@mstemm do you recall why we have a list here but a list pointer in `ruleset_filters::m_filter_by_event_tag`?
I don't want to change it in this PR but I'm curious...